wagtailcache/cache.py

```diff
@@ -51,6 +52,20 @@ def _patch_header(response: HttpResponse, status: Status) -> None:
         response[wagtailcache_settings.WAGTAIL_CACHE_HEADER] = status.value
 
 
+def _clean_uri(uri: str):
+    """
+    remove Trailingslash , remove protocol and subdomain and decode URI.
+    Reasons:
+            - clean keyring
+            - problems with reverse Proxy
+            - problems with special characters e.g. Ü,Ä,ö,ß
+    """
+    expr = r"https://www\.|http://www\.|https://www|http://|https://|www\."
```

Comment:
I'm not sure if removing the `www` is necessary/good here? Only because I would not want to risk caching the wrong thing if the user happens to have different content at the `www.example.com` and `example.com` websites (for example in a multi-site setup). Similar with the protocol.

It might be better to add a recommendation in the docs that they configure redirects for http/https and www at the web server level (Apache/NGINX) to maximize performance of the cache.

Reply:
I took another close look at the cache behaviour. We do not need this function only the `unquote`.